### ISSUE TITLE
Add GetString, GetInt, GetBool etc.

### ISSUE THREAD

**Issue by tflinkow** [flinkow/fluentmvvm#1]
Instead of only `Get<T>()`, introduce `GetString()`, `GetInt`, `GetBool` etc. for all common types.

### Pro:
+ Makes property getters easier to write

### Con:
- Crowds `FluentViewModelBase`'s public API

**Comment by tflinkow** [comment]
Introduce overloads not only for `Get`, but also for `Set`.

That in addition to correctly typing the generated backing fields (thus not `object`) should lead to better performance (eliminates boxing, casting, and reduces the number of branches in `GetValue` and `SetValue`.

Proposed API for both `FluentViewModelBase` and `IBackingFieldProvider`:

```csharp
T Get<T>(string fieldName);

short GetInt16(string fieldName);
int GetInt32(string fieldName);
long GetInt64(string fieldName);

string GetString(string fieldName);

// ...

bool Set<T>(string fieldName, T value);
    
bool Set(string fieldName, short value);
bool Set(string fieldName, int value);
bool Set(string fieldName, long value);

bool Set(string fieldName, string value);

// ...
```

The generated, non-generic implementations should only take into accounts the fields typed exactly like their return value / parameter, i.e. `GetInt32` should only check all fields typed `int` etc.

**Open questions:**

- what code should be emitted if no field of the specified type exists?
- can `where T : class` be used for the generic methods to avoid calling `Get<int>` instead of `GetInt32` etc. (thus introducing boxing)?
- if so, should we provide a `GetValueType` method for structures and enumerations?

**Comment by tflinkow** [comment]
[only if we cannot use generic constraints on `Get<T>` and `Set<T>`]

---

When implementing `Set<T>`, instead of emitting boxing instructions, e.g.

```il
box !!T
unbox.any [System.Private.CoreLib]System.Int32
```

which yields

```csharp
int casted = (int) (object) value; // value is of type T
```

can we not emit `conv.i4`?

---
Exceptions should only occur when the wrong opcode is used, which means the wrong `Set` overload was called - thus it's okay to fail.

**Open questions:**

- maybe better use `conv.ovf.x` instead of `conv.x`?
- should we throw on narrower types, i.e. should it be legal to pass `(short) 10` to an `int` field?
- what about `bool`, `char`, `(s)byte`, `(u)short`, and `uint`, which are all represented as `int` in CIL?

**Comment by tflinkow** [comment]
Implemented in 0.3.0-alpha
